lib/Serializer/functions.php

```diff
@@ -147,7 +147,7 @@ function repeatingElements(Writer $writer, array $items, string $childElementNam
  *
  * You can even mix the two array syntaxes.
  *
- * @param string|int|float|bool|array<int|string, mixed>|object $value
+ * @param string|int|float|bool|array<int|string, mixed>|object|null $value
```

Comment:
This function actually does let you pass `null` to it, and will just "do nothing" to the Writer object. In that case there is nothing to serialize but it does not complain about that.

Telling PHPstan that `null` is possible, stops it complaining about `elseif (is_object($value))` below - it now understands that `$value` could be `null`or an object at that point.

Reply:
What is the "good thing" to do for situations like this?
An input parameter allows some "complicated" set of inputs/types.
It is too complicated to be expressed in the actual PHP function declaration.
So at run-time PHP will actually let the caller pass anything in the parameter.
The function code checks the input parameter, doing "the expected thing" to process inputs of each of the "officially allowed" inputs.

The function code throws an exception if the passed-in value is not one of the "officially allowed" inputs. But PHPstan is going to consider that that is "useless" code.

Reply:
For this defensive style of programming you should set `treatPhpDocTypesAsCertain: false` in the PHPStan config

Reply:
If I do that, it will apply to all the code. So I just added the entry to `ignoreErrors`

Reply:
sure, if you are fine with ignoring the error thats fine, too.